app/src/main/java/com/example/recipeapp/Models/Parse/Taste.java

```diff
@@ -25,8 +88,30 @@ public void updateTaste(Recipe recipe, int numberOfVotes) {
      * @param newVal The value of the newly liked recipe
      * @param n How many votes user alreay have
      */
-    private void updateAverage(String key, float newVal, int n) {
+    private void updateAverage(String key, double newVal, int n) {
         double avg = getDouble(key);
         put(key, ((n - 1) * avg + newVal) / n);
     }
+
+    /**
+     * Calculates how different the new taste from the average taste of the user
+     * @param otherTaste Taste to compare
+     * @return
+     */
+    public Double calculateTasteDistance(Taste otherTaste) {
+        Double total = 0.0;
+        try {
+            fetchIfNeeded();
+        } catch (ParseException e) {
+            e.printStackTrace();
+        }
+        total += Recommendation.getNormalDistance(getDouble(KEY_SWEETNESS), otherTaste.sweetness) / NUMBER_OF_COMPONENTS;
+        total += Recommendation.getNormalDistance(getDouble(KEY_FATTINESS), otherTaste.fattiness) / NUMBER_OF_COMPONENTS;
+        total += Recommendation.getNormalDistance(getDouble(KEY_BITTERNESS), otherTaste.bitterness) / NUMBER_OF_COMPONENTS;
+        total += Recommendation.getNormalDistance(getDouble(KEY_SAVORINESS), otherTaste.savoriness) / NUMBER_OF_COMPONENTS;
+        total += Recommendation.getNormalDistance(getDouble(KEY_SOURNESS), otherTaste.sourness) / NUMBER_OF_COMPONENTS;
+        total += Recommendation.getNormalDistance(getDouble(KEY_SALTINESS), otherTaste.saltiness) / NUMBER_OF_COMPONENTS;
+        total += Recommendation.getNormalDistance(getDouble(KEY_SPICINESS), otherTaste.spiciness) / NUMBER_OF_COMPONENTS;
```

Comment:
Do we have a way to write to in a more scalable way? i.e. no need to update these lines if we add or remove a taste from the class

Reply:
discussed offline, we keep using the explicit class member first